Login page

atulmi opened the pull request:
The login page issues a popup with whatever error is passed into handleError(), popup doesn't look very pretty yet, but it is clearly visible.

Commit:
Update backend login function

atulmi committed:
Add handleError(error) function to login.jsx

atulmi committed:
Update login.jsx

Commit:
Update import

atulmi committed:
Added popup for failed login

MarkeyMarke commented on src/components/login.jsx:
```diff
@@ -82,27 +94,54 @@ class Login extends Component {
               type="submit"
               disabled={this.validate()}
-              className="btn btn-primary"
+              class="btn btn-primary"
```
If I remember correctly, JSX should NOT have any class tags, as it is already a keyword for components. Please replace this with className

MarkeyMarke commented on src/components/login.jsx:
```diff
@@ -57,21 +65,25 @@ class Login extends Component {
-    var tempThis = this; // Stores current value of this
-    var loginFunction = login(this.state.account.username.trim(),
-    this.state.account.password.trim());
+    var that = this; // Stores current value of this
```
I asked Dominic to change the names of these variables to add more clarity, so could someone please re-apply his changes? Please include his comments too!

MarkeyMarke reviewed (changes requested):
There's a few small tweaks I'd like to see before merging the changes.

atulmi committed:
Change button "class" to "className" in login.jsx

atulmi committed:
Change "Close" option in login failure popup to button

atulmi committed:
Added toggling for password visibility, and glyphicons to input fields

Also removed the popup for login failure. Instead, the error message will display under whichever input field the login error occurs in. This looks much cleaner.

atulmi committed:
Made password visibility toggling easier

Now, clicking anywhere in the password toggle box will perform visibility toggling, instead of having to click the icon itself.

Commit:
Refactor variable names
